### DISCUSSION TITLE
Easily run and connect to remote MCP servers with SuperGateway

### DISCUSSION THREAD

**Discussion by adamrb** [danny-avila/LibreChat#5141] [General]
For anyone looking for a more reliable way of running an MCP server for LibreChat...

I discovered https://github.com/supercorp-ai/supergateway, which provides a pretty straightforward way to run MCP servers over SSE.  It's easy to run inside Docker container with custom start commands. You could also do this in docker-compose.override.yaml, but I run it separate so I don't have to have Librechat depend on it. I found that Librechat won't retry a failed MCP server, so it needs to be up and running before Librechat starts.

Here's an example of a working docker-compose configuration (I use Portainer):

```services:
  supergateway-webresearch:
    image: node:23
    command: >
      sh -c "npm install -g @mzxrai/mcp-webresearch@latest &&
             npx playwright install-deps &&
             npx -y supergateway --stdio 'npx -y @mzxrai/mcp-webresearch@latest'"
    ports:
      - "8101:8000"
    restart: unless-stopped
    environment:
      - NODE_ENV=production
```

And in my librechat.yaml

```# MCP Server configuration
mcpServers:
  fetch:
    url: http://10.10.1.1:8100/sse
  webresearch:
    url: http://10.10.1.1:8101/sse
 ```

If you're running a python-based MCP server, you'll probably want to use: https://hub.docker.com/r/nikolaik/python-nodejs

**Comment by danny-avila** [comment]
Thanks for sharing!

**Comment by stanvx** [comment]
I've been using [mcp-gateway](https://github.com/acehoss/mcp-gateway) and find it performs well. I like that I can configure multiple MCP servers within a single container and differentiate each MCP Server with unique paths.

Are there any notable advantages of using [SuperGateway](https://github.com/supercorp-ai/supergateway) for LibreChat?
